GCC warns about sq.c because it includes a C++ header

For example:

```
/home/travis/build/paulsapps/alive/3rdParty/squirrel/include/sqconfig.h:11:23: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
 typedef unsigned long long SQHash; /*should be the same size of a pointer*/
                       ^
/home/travis/build/paulsapps/alive/3rdParty/squirrel/include/sqconfig.h:32:33: warning: C++ style comments are not allowed in ISO C90
 typedef __int64 SQRawObjectVal; //must be 64bits
```

Because sq.c is including sqconfig.h. What do you think of a PR to simply rename sq.c to sq.cpp so it builds and C++ and therefore "fixes" the warnings?

--- a/sq/CMakeLists.txt
+++ b/sq/CMakeLists.txt
@@ -1,4 +1,7 @@
 if(NOT DEFINED DISABLE_DYNAMIC)
+  if(CMAKE_COMPILER_IS_GNUCXX)
+    set_source_files_properties(sq.c PROPERTIES COMPILE_FLAGS -std=c99)
+  endif()
   add_executable(sq sq.c)
   set_target_properties(sq PROPERTIES LINKER_LANGUAGE C)
   target_link_libraries(sq squirrel sqstdlib)

